File: src/main/java/blog/dao/BlogMainDao.java
package blog.dao;

import blog.bean.BlogMainPojo;
import blog.bean.CommentPojo;
import blog.bean.WordPojo;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Map;

/**
 * Created by cao on 2017/1/10.
 */

@Repository("blog/dao/BlogMainDao")

public interface BlogMainDao {
    List<BlogMainPojo> getWord(Map map);

    List<WordPojo> toWord(int id);

    List<CommentPojo> getComments(int id);

    int putReply(Map parm);
}

File: src/main/kotlin/blog/dao/BlogMainDao.kt
package blog.dao

import blog.bean.BlogMainPojo
import blog.bean.CommentPojo
import blog.bean.MessagePojo
import blog.bean.Word
import org.apache.ibatis.annotations.Mapper
import java.util.*

/**
 * @author Frank
 * @date 2018/7/7 15:50
 */
@Mapper
interface BlogMainDao {

    fun getWord(map: Map<*, *>): List<BlogMainPojo>

    fun toWord(id: Int): List<Word>

    fun getComments(id: Int): List<CommentPojo>

    fun putReply(parm: Map<*, *>): Int

    fun getWordText(wordTextId: String): Map<*, *>

    fun getMessages(): List<MessagePojo>

    fun putMessage(nickname: String, date: Date, content: String, contactway: String): Int

}